features/eolearn/tests/test_feature_manipulation.py

```diff
@@ -43,6 +43,52 @@ def test_content_after_timefilter(self):
         self.assertEqual(new_interval, updated_interval)
         self.assertEqual(new_timestamps, updated_timestamps)
 
+    def test_value_fillout(self):
+        feature = (FeatureType.DATA, 'TEST')
+        shape = (8, 10, 10, 5)
+        data = np.random.randint(0, 100, size=shape).astype(np.float)
+        eopatch = EOPatch(data={'TEST': data})
+
+        self.assertRaises(ValueError, ValueFilloutTask, feature, operations='x')
+        self.assertRaises(ValueError, ValueFilloutTask, feature, operations=4)
+
+        self.assertRaises(ValueError, ValueFilloutTask.fill, None, operation='f')
+        self.assertRaises(ValueError, ValueFilloutTask.fill, np.zeros((4, 5)), operation='x')
+
+        # nothing to be filled, return the same eopatch object immediately
+        eopatch_new = ValueFilloutTask(feature, operations='fb', axis=0)(eopatch)
+        self.assertEqual(eopatch, eopatch_new)
+
+        eopatch[feature][0, 0, 0, :] = np.nan
+
+        # filling forward temporally should not fill nans
+        eopatch = ValueFilloutTask(feature, operations='f', axis=0)(eopatch)
+        self.assertTrue(np.isnan(eopatch[feature][0, 0, 0, :]).all())
```

Comment:
A few of these tests could be improved. Here `ValueFilloutTask` shouldn't change the feature at all. So you could even check that an array before this task and an array after it are completely the same. Similarly you could do in the next test. On the other hand you also might have to check that you don't compare the same object. If I'm not mistaken the task creates a new array object in the process.